test/contrib/test_tfp.py

```diff
@@ -17,71 +16,50 @@
 from numpyro.infer.reparam import TransformReparam
 
 
-# XXX: for some reasons, pytest raises ImportWarning when we import tfp
-@pytest.mark.filterwarnings("ignore:can't resolve package")
-def test_api_consistent():
-    from numpyro.contrib.tfp import distributions as tfd
-
-    for name in tfd.__all__:
-        if name in numpyro.distributions.__all__:
-            tfp_dist = getattr(tfd, name)
-            numpyro_dist = getattr(numpyro.distributions, name)
-            if type(numpyro_dist).__name__ == "function":
-                numpyro_dist = getattr(numpyro.distributions, name + "Logits")
-            for p in tfp_dist.arg_constraints:
-                assert p in inspect.getfullargspec(tfp_dist.__init__)[0]
-
-
 def test_dist_pytree():
-    from numpyro.contrib.tfp import distributions as tfd
-
-    def f(x):
-        return tfd.Normal(x, 1)
+    from tensorflow_probability.substrates.jax import distributions as tfd
 
-    jit(f)(0)
+    from numpyro.contrib.tfp.distributions import TFPDistribution
 
+    @jit
+    def f(x):
+        with numpyro.handlers.trace() as tr:
+            numpyro.sample("x", tfd.Normal(x, 1))
+        return tr["x"]["fn"]
 
-@pytest.mark.filterwarnings("ignore:can't resolve package")
-def test_independent():
-    from numpyro.contrib.tfp import distributions as tfd
+    with numpyro.handlers.seed(rng_seed=0):
```

Comment:
nit: I think it is a better practice to move seed handler inside the `@jit`. Otherwise, if you run
```
with numpyro.handlers.seed(rng_seed=1):
    res = f(0.0)
```
you will get the same result as `rng_seed=0`.